apiservers/engine/backends/container.go

```diff
@@ -445,3 +607,67 @@ func (c *Container) imageExist(imageID string) (storeName string, err error) {
 
 	return host, nil
 }
+
+func (c *Container) getImageMetadataFromStoragePL(image string) (*viccontainer.VicContainer, error) {
+	// FIXME: This is a temporary workaround until we have a name resolution story.
+	// Call imagec with -resolv parameter to learn the name of the vmdk and put it into in-memory map
+	cmdArgs := []string{"-reference", image, "-resolv", "-standalone", "-destination", os.TempDir()}
+
+	out, err := exec.Command(Imagec, cmdArgs...).Output()
+	if err != nil {
+		log.Printf("%s exit code: %s", Imagec, err)
+		return nil,
+			derr.NewErrorWithStatusCode(fmt.Errorf("Container look up failed"),
+				http.StatusInternalServerError)
+	}
+	var v1 struct {
+		ID string `json:"id"`
+		// https://github.com/docker/engine-api/blob/master/types/container/config.go
+		Config container.Config `json:"config"`
+	}
+	if err := json.Unmarshal(out, &v1); err != nil {
+		return nil,
+			derr.NewErrorWithStatusCode(fmt.Errorf("Failed to unmarshall image history: %s", err),
+				http.StatusInternalServerError)
+	}
+	log.Printf("v1 = %+v", v1)
+
+	imageMetadata := &viccontainer.VicContainer{
+		ID:     v1.ID,
+		Config: &v1.Config,
+	}
+
+	return imageMetadata, nil
+}
+
+func (c *Container) getContainerConfigFromExecPL(name string) (*container.Config, error) {
+	// Get an API client to the portlayer
+	client := PortLayerClient()
```

Comment:
We're not initializing the client lazily and this doesn't perform additional checking. Would be cleaner to use a global var directly and avoid the unreachable nil check.